frontend/lib/src/components/elements/DeckGlJsonChart/DeckGlJsonChart.tsx

```diff
@@ -14,288 +14,90 @@
-        ) {
-          body = body.replace(match, info.object.properties[variable])
-        }
-      })
-    }
-    return body
-  }
-
-  onViewStateChange: DeckProps["onViewStateChange"] = ({ viewState }) => {
-    this.setState({ viewState })
-  }
-
-  render(): ReactNode {
-    const deck = DeckGlJsonChart.getDeckObject(this.props, this.state)
-    const { viewState } = this.state
-    const { width } = this.props
-
-    return (
-      <StyledDeckGlChart
-        className="stDeckGlJsonChart"
-        data-testid="stDeckGlJsonChart"
-        width={width}
+    setIsInitialized(true)
+  }, [])
+
+  return (
+    <StyledDeckGlChart
+      className="stDeckGlJsonChart"
+      data-testid="stDeckGlJsonChart"
+      width={width}
+      height={deck.initialViewState.height}
+    >
+      <DeckGL
+        viewState={viewState}
+        onViewStateChange={onViewStateChange}
         height={deck.initialViewState.height}
+        width={width}
+        layers={isInitialized ? deck.layers : EMPTY_LAYERS}
+        getTooltip={createTooltip}
+        // @ts-expect-error There is a type mismatch due to our versions of the libraries
+        ContextProvider={MapContext.Provider}
+        controller
       >
-        <DeckGL
-          viewState={viewState}
-          onViewStateChange={this.onViewStateChange}
+        <StaticMap
           height={deck.initialViewState.height}
           width={width}
-          layers={this.state.initialized ? deck.layers : []}
-          getTooltip={this.createTooltip}
-          // @ts-expect-error There is a type mismatch due to our versions of the libraries
-          ContextProvider={MapContext.Provider}
-          controller
-        >
-          <StaticMap
-            height={deck.initialViewState.height}
-            width={width}
-            mapStyle={
-              deck.mapStyle &&
-              (typeof deck.mapStyle === "string"
-                ? deck.mapStyle
-                : deck.mapStyle[0])
-            }
-            mapboxApiAccessToken={
-              this.props.element.mapboxToken || this.props.mapboxToken
-            }
+          mapStyle={
+            deck.mapStyle &&
+            (typeof deck.mapStyle === "string"
+              ? deck.mapStyle
+              : deck.mapStyle[0])
+          }
+          mapboxApiAccessToken={props.element.mapboxToken || props.mapboxToken}
+        />
+        <StyledNavigationControlContainer>
+          <NavigationControl
+            data-testid="stDeckGlJsonChartZoomButton"
+            showCompass={false}
           />
-          <StyledNavigationControlContainer>
-            <NavigationControl
-              data-testid="stDeckGlJsonChartZoomButton"
-              showCompass={false}
-            />
-          </StyledNavigationControlContainer>
-        </DeckGL>
-      </StyledDeckGlChart>
-    )
-  }
+        </StyledNavigationControlContainer>
+      </DeckGL>
+    </StyledDeckGlChart>
+  )
 }
 
 export default withTheme(
```

Comment:
nit: I think for functional components the `useTheme` hook is preferred. I think the `withTheme` is more of a historical relict used by class-based components.

```typescript
const theme: EmotionTheme = useTheme()
```

Reply:
Updated!